test/functional/test_framework/test_node.py

```diff
@@ -210,6 +213,16 @@ def start(self, extra_args=None, *, cwd=None, stdout=None, stderr=None, env=None
         if extra_args is None:
             extra_args = self.extra_args
 
+        # If listening and no -bind is given, then bitcoind would bind P2P ports on
+        # 0.0.0.0:P and 127.0.0.1:18445 (for incoming Tor connections), where P is
+        # a unique port chosen by the test framework and configured as port=P in
+        # bitcoin.conf. Don't bind on 127.0.0.1:18445 because it is not needed and
+        # to avoid collisions.
+        will_listen = all(e != "-nolisten" and e != "-listen=0" for e in extra_args)
+        has_explicit_bind = self.has_explicit_bind or any(e.startswith("-bind=") for e in extra_args)
+        if will_listen and not has_explicit_bind:
+            extra_args.append("-bind=0.0.0.0")
```

Comment:
All these

https://github.com/bitcoin/bitcoin/blob/b1ba1b178f501daa1afdd91f9efec34e5ec1e294/test/functional/test_framework/util.py#L393-L423

can be considered as divergence from the default `bitcoind` behavior.

The suggested change adds a new `bitcoind` config option just for the test framework, I do not like that. I guess it may be possible to avoid Tor collisions by spreading the ports without using a new config option, by using `-bind=127.0.0.1:torport=onion`, but then, I do not think that will be simpler than the current approach and and also, as stated in the comment above "Don't bind on 127.0.0.1:18445 because it is not needed and to avoid collisions." avoiding collisions is part of the reason why not to bind. The other reason is that "it is not needed". I would rather avoid the bind if is not needed, instead of trying to accommodate the it.

What do other reviewers think? Leave it like now, or try to bind on the tor port in a non-conflicting way even if not needed? @achow101, @pinheadmz, @laanwj, @stratospher, @jonatack?

Reply:
> I'm actually in favor of adding tor_port() to util.py and binding every bitcoind in the functional tests to a unique port by default.

Done, see if you like it. No new `bitcoind` config options introduced.